tree/dataframe/test/dataframe_snapshot.cxx

```diff
@@ -1455,4 +1455,67 @@ TEST(RDFSnapshotMore, ZeroOutputEntriesMT)
    gSystem->Unlink(fname);
 }
 
+// Test for custom basket size in Snapshot, ROOT - #17418
+TEST(RDFSnapshotMore, CustomBasketSize)
+{
+    // Create a simple ROOT file with a TTree
+    auto input_file = TFile::Open("input_file.root", "RECREATE");
+    auto tree = new TTree("tree", "Test Tree");
```

Comment:
Do not create the dataset like this. Prefer RAII struct such as the ones you can find in many RDF tests. An example is https://github.com/root-project/root/blob/8b3840f73be1d892334ba7df6d2867cad470c196/tree/dataframe/test/dataframe_vary.cxx#L1626